File: samples/Python/exponentiation-operator.py
>>> import operator
>>> class num(int):
    def __pow__(self, b):
        print "Empowered"
        return operator.__pow__(self+0, b)


>>> x = num(3)
>>> x**2
Empowered
9
>>> class num(float):
    def __pow__(self, b):
        print "Empowered"
        return operator.__pow__(self+0, b)


>>> x = num(2.5)
>>> x**2
Empowered
6.25
>>>

File: samples/Lua/exponentiation-operator.lua
number = {}

function number.pow( a, b )
    local ret = 1
    if b >= 0 then
        for i = 1, b do
            ret = ret * a.val
        end
    else
        for i = b, -1 do
            ret = ret / a.val
        end
    end
    return ret
end

function number.New( v )
    local num = { val = v }
    local mt = { __pow = number.pow }
    setmetatable( num, mt )
    return num
end

x = number.New( 5 )
print( x^2 )                   --> 25
print( number.pow( x, -4 ) )   --> 0.016
